nvflare/edge/assessors/model_update.py

```diff
@@ -116,6 +240,38 @@ def _do_child_update(self, update: Shareable, fl_ctx: FLContext) -> (bool, Optio
         # Update available devices
         if report.available_devices:
             self.device_manager.update_available_devices(report.available_devices, fl_ctx)
+            # Reset wait timer if we now have enough devices
+            if self.device_wait_start_time is not None and self.device_manager.has_enough_devices(fl_ctx):
+                self.device_wait_start_time = None
+                self.should_stop_job = False  # Reset stop job flag
+                # Stop timeout tracking since we have enough devices
+                self._stop_timeout_tracking()
```

Comment:
Suggest using a single long-lived thread with a wait flag, instead of repeatedly starting and stopping a thread.